hawkular-command-gateway-war/src/main/java/org/hawkular/cmdgw/command/ws/WsSessions.java

```diff
@@ -199,6 +200,20 @@ public Session getSession(String key) {
     }
 
     /**
+     * Copies the {@link Session}s stored inside {@link #sessions} into a new {@link ArrayList} and returns the copy.
+     *
+     * @return a new muttable {@link ArrayList}
+     */
+    public List<Session> getSessions() {
+        sessionsLockRead.lock();
+        try {
+            return sessions.values().stream().map(SessionEntry::getSession).collect(Collectors.toList());
```

Comment:
Keeping aside that `sessions` is not a collection of `Session` s (it is a map having the sessions buried somewhere in the values and it therefore needs to be transformed), `new ArrayList<>(sessions)` would not protect me against concurrent removals and additions inside `sessions`, would it?